src/main/java/net/dv8tion/jda/api/events/message/react/GenericMessageReactionEvent.java

```diff
@@ -60,17 +60,23 @@ public GenericMessageReactionEvent(@Nonnull JDA api, long responseNumber, @Nulla
     /**
      * The id for the user who owns the reaction.
      *
+     * This is null if the user was not cached.
+     *
      * @return The user id
      */
-    @Nonnull
+    @Nullable
     public String getUserId()
     {
+        if (userId == 0)
+            return null;
         return Long.toUnsignedString(userId);
     }
```

Comment:
afaict, this isn't true.
The `userId` is never `0` here because it is provided by the gateway event and passed in:
https://github.com/DV8FromTheWorld/JDA/blob/12d4a0af25cc6bad3a8f4a37bd5230d861e66268/src/main/java/net/dv8tion/jda/internal/handle/MessageReactionHandler.java#L59
https://github.com/DV8FromTheWorld/JDA/blob/12d4a0af25cc6bad3a8f4a37bd5230d861e66268/src/main/java/net/dv8tion/jda/internal/handle/MessageReactionHandler.java#L155-L158